app/controllers/publishers/site_banners_controller.rb

```diff
@@ -21,12 +21,23 @@ def create
   end
 
   def update_logo
+    if params[:image].length > MAX_IMAGE_SIZE
+      # (Albert Wang): We should consider supporting alerts. This might require a UI redesign
+      # alert[:error] = "File size too big!"
+      head :payload_too_large and return
+    end
     site_banner = current_publisher.site_banner
     update_image(site_banner.logo)
     head :ok
   end
 
   def update_background_image
+    if params[:image].length > MAX_IMAGE_SIZE
+      # (Albert Wang): We should consider supporting alerts. This might require a UI redesign
+      # alert[:error] = "File size too big!"
+      head :payload_too_large and return
+    end
+
```

Comment:
I'm not aware of a way to do `render and return` from a subroutine.

Reply:
Highlighted the wrong code

```
  def update_logo
    if params[:image].length > MAX_IMAGE_SIZE
      # (Albert Wang): We should consider supporting alerts. This might require a UI redesign
      # alert[:error] = "File size too big!"
      head :payload_too_large and return
    end
    site_banner = current_publisher.site_banner
    update_image(site_banner.logo)
    head :ok
  end

  def update_background_image
    if params[:image].length > MAX_IMAGE_SIZE
      # (Albert Wang): We should consider supporting alerts. This might require a UI redesign
      # alert[:error] = "File size too big!"
      head :payload_too_large and return
    end

    site_banner = current_publisher.site_banner
    update_image(site_banner.background_image)
    head :ok
  end
```
have you considered one general function here?

Reply:
not possible since it's meant to have a dedicated route.